File: Data_Structures/stack.py
```
from array import array


class Stack:

    def __init__(self):
        self.__count = 0
        self.__size = 5
        self.__main_array = array("i", [0 for _ in range(self.__size)])

    def __str__(self):
        return str(self.__main_array[:self.__count])[11:-1]

    def common_actions(self, array_1):
        array_2 = array("i", [0 for _ in range(self.__size)])
        for index in range(self.__count):
            array_2[index] = array_1[index]
        return array_2

    def push(self, element):
        if self.__count < self.__size:
            self.__main_array[self.__count] = element
            self.__count += 1
        else:
            helper_array = self.common_actions(self.__main_array)
            self.__size += 5
            self.__main_array = self.common_actions(helper_array)
            self.__main_array[self.__count] = element
            self.__count += 1

    def pop(self):
        target_element = self.__main_array[self.__count - 1]
        self.__count -= 1
        self.__main_array = self.common_actions(self.__main_array)
        return target_element

    def peek(self):
        return self.__main_array[self.__count - 1]

    def to_list(self):
        return [self.__main_array[index] for index in range(self.__count)]  # აქ self.__main_array[self.__count - 1]-ის
        # დაბრუნებაც შეიძლებოდა, მაგრამ ვიზუალური კუთხით ესე რო მიუვიდეს იუზერს უკეთესია.

    def is_empty(self):
        return self.__count == 0

    def clear(self):
        self.__count = 0
        self.__size = 5
        self.__main_array = array("i", [0 for _ in range(self.__size)])

    def show(self):
        return self.__str__()


if __name__ == '__main__':
    stack = Stack()

    for number in range(12):
        stack.push(number)

    print(stack)
    print(stack.pop())
    print(stack.peek())
    print(stack.is_empty())
    print(stack.peek())
    print(stack)
    listed_stack = stack.to_list()
    print(listed_stack)
    listed_stack.append(123)
    print(listed_stack)
    
```

File: Data_Structures/Stack.java
public class Stack {
    private int count = 0;
    private int size = 5;
    private int[] mainArray = new int[this.size];

    protected int[] commonActions(int[] array_1) {
        int[] array_2 = new int[this.size];
        for (int index = 0; index < this.count; index ++) {
            array_2[index] = array_1[index];
        }
        return array_2;
    }

    protected void push(int element) {
        if (this.count < this.size) {
            this.mainArray[this.count] = element;
            this.count ++;
        }
        else {
            int[] helperArray = this.commonActions(this.mainArray);
            this.size += 5;
            this.mainArray = this.commonActions(helperArray);
            this.mainArray[this.count] = element;
            this.count ++;
        }
    }

    protected int pop() {
        int targetElement = this.mainArray[this.count - 1];
        this.count --;
        this.mainArray = this.commonActions(this.mainArray);

        return targetElement;
    }

    protected int peek() {
        return this.mainArray[this.count - 1];
    }

    protected int[] toList() {
        return this.mainArray;
    }

    protected boolean isEmpty() {
        return this.count == 0;
    }

    protected void clear() {
        this.count = 0;
        this.size = 5;
        this.mainArray = new int[this.size];
    }

    protected int length() {
        return this.count - 1;
    }

    protected void print() {
        for (int index = 0; index < this.count; index ++) {
            System.out.print(this.mainArray[index] + " ");
        }
    }


    public static void main(String[] args) {
        Stack stack = new Stack();
        for (int i = 0; i < 12; i ++) {
            stack.push(i);
        }
        stack.print();
        System.out.println();
        stack.toList();
        System.out.println(stack.pop());
        System.out.println();
        stack.print();
        stack.push(15);
        System.out.println(stack.peek());
        System.out.println(stack.isEmpty());
        System.out.println(stack.length());
        stack.clear();
        stack.print();
    }
}
